### ISSUE TITLE
unfree packages can be installed by default and display a warning

### ISSUE THREAD

**Issue by mkenigs** [flox/flox#492]
# Acceptance Criteria

**Given** I have an environment freshly `flox init`
**When** I `flox install vscode` (unfree package example)
**Then** I see a message...

```
⚠️ The package 'vscode' is not free. Please verify the licensing terms of use.
✅ 'vscode' installed to environment name at /Users/ghudgins/name

```

_Don't put unfree in manifest by default_
**And When** I `flox edit`
**Then** I do not see any options related to unfree (we assume lack of the option means unfree is true)

_setting allow unfree still works_
**And When** I add `options.allow.unfree = false`
**And** I `flox install spotify`
**Then** I see an error not allowing installation

```
❌ ERROR: Failed to build environment.

Package 'spotify' is marked as unfree.

Allow unfree packages by setting 'options.allow.unfree = true' in manifest.toml
``` 

_Flox search honors the settings of the environment (including implicit defaults)_
**Given** an environment has no options set
**When** I `flox search vscode` <-- unfree package
**Then** I see results
**And When** I `flox search CHOWTape` <-- broken package
**Then** I don't see results

# Notes 
Trying to install an unfree package prints an ugly Nix error. We should either catch the error and print something nicer or enable unfree packages by default.

```
> flox install vscode
ERROR: error building environment: warning: unknown setting 'trusted-users'
    error:
           … in the condition of the assert statement

             at /nix/store/l45hwj0x8rpmz64qli38m2j831n82zkw-source/lib/customisation.nix:229:17:

              228|       drvPath = assert condition; drv.drvPath;
              229|       outPath = assert condition; drv.outPath;
                 |                 ^
              230|     };

           … while evaluating the attribute 'handled'

             at /nix/store/l45hwj0x8rpmz64qli38m2j831n82zkw-source/pkgs/stdenv/generic/check-meta.nix:447:7:

              446|       # or, alternatively, just output a warning message.
              447|       handled =
                 |       ^
              448|         {

           (stack trace truncated; use '--show-trace' to show the full trace)

           error: Package ‘vscode-1.78.2’ in /nix/store/l45hwj0x8rpmz64qli38m2j831n82zkw-source/pkgs/applications/editors/vscode/vscode.nix:59 has an unfree license (‘unfree’), refusing to evaluate.

           a) To temporarily allow unfree packages, you can use an environment variable
              for a single invocation of the nix tools.

                $ export NIXPKGS_ALLOW_UNFREE=1

            Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
            (Flake) command, `--impure` must be passed in order to read this
            environment variable.

           b) For `nixos-rebuild` you can set
             { nixpkgs.config.allowUnfree = true; }
           in configuration.nix to override this.

           Alternatively you can configure a predicate to allow specific packages:
             { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
                 "vscode"
               ];
             }

           c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
             { allowUnfree = true; }
           to ~/.config/nixpkgs/config.nix.
```

Some prior discussion is in https://github.com/flox/product/issues/125

**Comment by aakropotkin** [comment]
TODO: create ticket with technical approach

**Comment by mkenigs** [comment]
There's other packages like ruby that fail to install because of `NIXPKGS_ALLOW_INSECURE` (e.g. `flox install ruby_2_7`). Not sure if we'll solve that as part of this issue or need to open a separate issue.

**Comment by ghudgins** [comment]
would need a global place to set this and declare it in the manifest. we should update the template

**Comment by ghudgins** [comment]
consider warnings for licensing

**Comment by ghudgins** [comment]
absent some other design, I suggest something like this:


`$ flox install paid-package`

⚠️ 'paid-package' installed to environment temp at /Users/ghudgins/temp.
This package is not free. 
Please ensure you have licensed paid-package according to the terms of use.

**Comment by mkenigs** [comment]
We may also want to add a warning that the package must be built from source, so it could take a while

**Comment by aakropotkin** [comment]
In progress : https://github.com/flox/flox/pull/714

**Comment by zmitchell** [comment]
We'll be handling this as part of the rules engine, so I'm adding it to that epic

**Comment by mkenigs** [comment]
Blocked by https://github.com/flox/flox/issues/762 so moving to backlog

**Comment by ghudgins** [comment]
this will close as part of the rules engine epic. keeping it for tracking purposes

**Comment by ghudgins** [comment]
do before we do https://github.com/flox/product/issues/607

**Comment by ghudgins** [comment]
+1 from https://discourse.flox.dev/t/some-notes-for-people-migrating-to-flox-0-9-0/830

**Comment by ghudgins** [comment]
verified 🚀

**Comment by ghudgins** [comment]
re-opening as we need small adjustments to the man pages for `manifest.toml`

**Comment by ghudgins** [comment]
can't remember. will open later tickets if necessary
